src/Interpreter.hs

```diff
@@ -32,5 +39,20 @@ close = GhciWrapper.close . sessionInterpreter
 reload :: Session -> IO String
 reload Session{..} = evalEcho sessionInterpreter ":reload"
 
-hspec :: Session -> IO String
-hspec Session{..} = evalEcho sessionInterpreter $ "System.Environment.withArgs " ++ (show $ "--color" : sessionHspecArgs) ++ " $ Test.Hspec.hspec spec"
+data Summary = Summary {
+  summaryExamples :: Int
+, summaryFailures :: Int
+} deriving (Eq, Show, Read)
+
+hspec :: Session -> IO (String, Maybe Summary)
+hspec Session{..} = do
+  r <- evalEcho sessionInterpreter $ "System.Environment.withArgs " ++ (show $ "--color" : sessionHspecArgs) ++ " $ Test.Hspec.Runner.hspecResult spec"
+  return $ case reverse $ lines r of
+    x : _ | Just summary <- readMaybe (dropAnsiEscapeSequences x) -> (r, Just summary)
+    _ -> (r, Nothing)
+  where
+    dropAnsiEscapeSequences xs
+      | "Summary" `isPrefixOf` xs = xs
+      | otherwise = case xs of
+          _ : ys -> dropAnsiEscapeSequences ys
+          [] -> []
```

Comment:
Yes, we talked about this offline. `hspec`s output includes the closing escape sequence (to reset to the default color) after the newline of the previous line. So it has to be stripped here. And just dropping everything till the sequence `"Summary"` seems to be the easiest.